bon-macros/src/lib.rs

```diff
@@ -100,3 +102,93 @@ pub fn bon(params: TokenStream, item: TokenStream) -> TokenStream {
         .unwrap_or_else(|err| error::error_into_token_stream(err, item.into()))
         .into()
 }
+
+/// Similar to [`maplit::hashmap!`]/[`maplit::btreemap!`] but converts each key and value with
+/// [`Into::into()`].
+///
+/// There are no separate variants for [`HashMap`] and [`BTreeMap`] since the macro works with any
+/// type that implements [`FromIterator<(K, V)>`].
+///
+/// The macro also performs rudimentary uniqueness checking on keys: Syntactically equal keys are
+/// rejected with a compile error.
+///
+/// A good example of the use case for this macro is when you want to create a
+/// `HashMap<String, String>` where part of the keys or values are hardcoded string literals of type `&str`
+/// and the other part is made of dynamic [`String`] values.
+///
+/// ```rust
+/// # use bon_macros as bon;
+/// # use std::collections::HashMap;
+/// fn convert_media(input_extension: &str, output_extension: &str) -> std::io::Result<()> {
+///     let ffmpeg_args: HashMap<String, String> = bon::map!{
+///         "-i": format!("input.{input_extension}"),
+///         "-y": format!("output.{output_extension}"),
+///     };
+///
+///     let mut command = std::process::Command::new("ffmpeg");
+///     for (flag, value) in ffmpeg_args {
+///         command.arg(flag).arg(value);
+///     }
+///     command.output()?;
+///
+///     Ok(())
+/// }
+/// ```
+///
+/// [`BTreeMap`]: std::collections::BTreeMap
+/// [`HashMap`]: std::collections::HashMap
+/// [`maplit::hashmap!`]: https://docs.rs/maplit/latest/maplit/macro.hashmap.html
+/// [`maplit::btreemap!`]: https://docs.rs/maplit/latest/maplit/macro.btreemap.html
+#[proc_macro]
+pub fn map(input: TokenStream) -> TokenStream {
+    let entries = syn::parse_macro_input!(input with util::parse_map_macro_input);
+
+    map::generate(entries)
+        .unwrap_or_else(|err| err.write_errors())
+        .into()
+}
+
+/// Similar to [`maplit::hashset!`]/[`maplit::btreeset!`] but converts each key and value with
+/// [`Into::into()`].
+///
+/// There are no separate variants for [`HashSet`] and [`BTreeSet`] since the macro works with any
+/// type that implements [`FromIterator<T>`].
+///
+/// The macro also performs rudimentary uniqueness checking on keys: Syntactically equal keys are
+/// rejected with a compile error.
+///
+/// A good example of the use case for this macro is when you want to create a
+/// `Hashset<String, String>` where part of the keys or values are hardcoded string literals of type `&str`
+/// and the other part is made of dynamic [`String`] values.
+///
+/// ```rust
+/// # use bon_macros as bon;
+/// # use std::collections::HashSet;
+/// fn convert_media(input_extension: &str, output_extension: &str) -> std::io::Result<()> {
+///     let ffmpeg_args: HashSet<String> = bon::set!{
+///         "-i",
+///         format!("input.{input_extension}"),
+///         "-y",
+///         format!("output.{output_extension}"),
+///     };
```

Comment:
This example is also not correct. `HashSet` produces elements in arbitrary order. It doesn't make sense to put CLI args in a set anyway.

Reply:
Similar synthetic example.